internal/app/service/storage.go

```diff
@@ -0,0 +1,90 @@
+package service
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+
+	"github.com/MrTomSawyer/url-shortener/internal/app/models"
+)
+
+type Storage struct {
+	path        string
+	largestUUID int
+}
+
+func (s *Storage) Write(uj *models.URLJson) error {
+	file, err := os.OpenFile(s.path, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0644)
+	if err != nil {
+		return fmt.Errorf("error opening file to write: %v", err)
+	}
+	defer file.Close()
+
+	fileWriter := bufio.NewWriter(file)
+	defer fileWriter.Flush()
+
+	parsedURLJSON, err := json.Marshal(uj)
+	if err != nil {
+		return fmt.Errorf("error parsing data: %v", err)

```

Comment:
```suggestion
		return fmt.Errorf("error parsing data: %w", err)
```
здесь и в остальных местах, кажется писал об этом - w позволяет распаковать ошибку

Reply:
done